fix(*): Style updates, add edit issue body functionality

housseindjirdeh opened the pull request:
Various fixes/updates for #381 

- Changes color of mark all notifications button

![image](https://user-images.githubusercontent.com/12476932/31062274-5bf9f83e-a6f7-11e7-86e0-b0179f43042f.png)

- Adds padding for no notifications message in notifications screen

![image](https://user-images.githubusercontent.com/12476932/31062279-7226bcc8-a6f7-11e7-901e-61f90f9b9479.png)

- Fixes this problem:

![image](https://user-images.githubusercontent.com/12476932/31062288-854ac254-a6f7-11e7-8881-fa1158362f4d.png)

- Allows editing of issue description

![editdescription](https://user-images.githubusercontent.com/12476932/31062297-9a0ee760-a6f7-11e7-996e-717d74c1dca9.gif)

- Makes title/subtitle of `IssueDescriptionListItem` better in terms of boldness

![image](https://user-images.githubusercontent.com/12476932/31062315-b9ef6262-a6f7-11e7-8f6e-cf695645f2fa.png)

- Removes the user title from repository list item (@lex111 - with this it's unlikely we'll see the `private` box being squeezed anymore as most repo titles aren't nearly as long)

![image](https://user-images.githubusercontent.com/12476932/31062366-2ae03046-a6f8-11e7-996a-e1cdb2ff2507.png)

housseindjirdeh commented on src/issue/issue.action.js:
```diff
@@ -172,6 +173,28 @@ export const editIssueComment = (issueCommentId, owner, repoName, body) => {
   };
 };
 
+export const editIssueBody = (owner, repoName, issueNum, body) => {
```
`editIssue` currently already exists and I'm definitely not a fan of how I'm passing `updateParams` to it. Because the comment is HTML, can't leverage that to pass in a new update parameter for the text to replace state. 

Will need to refactor how `editIssue` works later without a doubt and instead of passing in updateParams, just get what's returned from the API (and we can then just remove this)

lex111 commented:
> Removes the user title from repository list item (@lex111 - with this it's unlikely we'll see the private box being squeezed anymore as most repo titles aren't nearly as long)

Let's remove this label and display this [icon](https://octicons.github.com/icon/lock/)?

lex111 commented:
> Adds padding for no notifications message in notifications screen

And why are they needed? There in fact there is an alignment on the center, and after addition padding, any more will not be absolutely on the center to be displayed.

chinesedfan commented:
@housseindjirdeh User titles are needed, at least for searched result or stared repositories. This feature is added by #254 in particular. And I also mentioned the redundancy, but you missed it. :)

machour commented:
Seconding @lex111, let's use that emoji and bring back username/reponame :

If I'm on some user profile, then check his repos, I'll have to remember the user name, or navigate back and forth to have the repo full name to clone it

lex111 commented on src/components/repository-list-item.component.js:
```diff
@@ -63,7 +63,7 @@ const renderTitle = repository =>
       <View style={styles.titleWrapper}>
         <Text style={styles.title}>
-          {repository.full_name}
+          {repository.name}
```
As correctly noted, displaying only the name of the repository will not be convenient, so we need to add a check: if this is an auth screen, then show only the name of the repository, and if the user profile, then display the full name.

lex111 replied:
Although, maybe it can just add the `lock` icon to the end of the title? 
@chinesedfan so in fact will work?

chinesedfan replied:
Yes, it doesn't matter the pending element is the private box or the lock icon, as long as the parent element is Text and the child view has width&height specified.

housseindjirdeh replied:
Done

jouderianjr commented:
@housseindjirdeh What do you think about handling with edit issue body in another screen. Maybe we could create an `edit issue` for edit the description and the title.

housseindjirdeh commented:
> And why are they needed? There in fact there is an alignment on the center, and after addition padding, any more will not be absolutely on the center to be displayed.

Honestly I just felt that the message looks a bit squeezed for English copy?

![image](https://user-images.githubusercontent.com/12476932/31084936-c2fe6c9c-a763-11e7-9e09-2e2443149f3a.png)

Probably just imagining things, reverted it back and removed that padding :)

> User titles are needed, at least for searched result or stared repositories. This feature is added by #254 in particular. And I also mentioned the redundancy, but you missed it. :)

Have no idea how I missed that point. Thank you, and I completely forgot that `search` and `starred` repos would have to show that. Updated it so now only the `authenticated` list of users repositories won't show (otherwise it will :))

> Let's remove this label and display this icon?

Done:

![image](https://user-images.githubusercontent.com/12476932/31085036-1683323a-a764-11e7-89b2-b9b7037c7fb2.png)

housseindjirdeh commented:
@jouderianjr so you make a good point, and right now if you head to the settings screen you can modify a few issue settings (but not title or description)

![image](https://user-images.githubusercontent.com/12476932/31085236-949eb72a-a764-11e7-95ba-36cee7bd3c5f.png)

We could have issue title and description modifiable here but honestly I actually prefer at least the description being modified the same way as any other comment. It feels intuitive as I still think of it as a comment.

Definitely curious if you think there's a better option however, as well as how we can add functionality to edit the issue title in the future 🤔

lex111 commented:
> Honestly I just felt that the message looks a bit squeezed for English copy?

Oops, did not notice if padding left and right, then this should not affect the alignment, although at first glance it seemed to me that this is so. Excuse me, let's it return, but can we make them smaller?

lex111 commented:
> Let's remove this label and display this icon?

Do not you think that the size is big? In my opinion it was better if you reduce the size of the icon?

housseindjirdeh commented:
> Oops, did not notice if padding left and right, then this should not affect the alignment, although at first glance it seemed to me that this is so. Excuse me, let's it return, but can we make them smaller?

Cheers no worries. Added some padding but reduced it a bit :)

![image](https://user-images.githubusercontent.com/12476932/31086536-190a9706-a768-11e7-956f-38f633d8b065.png)

> Do not you think that the size is big? In my opinion it was better if you reduce the size of the icon?

And nice you're probably right. Reduced it:

![image](https://user-images.githubusercontent.com/12476932/31086564-28e595ea-a768-11e7-8080-522db9d96506.png)

machour commented:
@housseindjirdeh I'd rather use type="default" (or no type) for the "Mark all as read" button.
Let's keep "success" for **strong** actions, like merge

housseindjirdeh commented:
> @housseindjirdeh I'd rather use type="default" (or no type) for the "Mark all as read" button.
Let's keep "success" for strong actions, like merge

Done.

Although the default button looks interesting 🤔 Are these the intended colors? (just showing it defaulted for the merge button for demo, don't have any notifications in the moment that's why :))

![image](https://user-images.githubusercontent.com/12476932/31086812-e6bb1496-a768-11e7-862f-8f915f19245c.png)

machour commented:
@housseindjirdeh that icon should be black, otherwise it's the intended look.

jouderianjr commented:
> We could have issue title and description modifiable here but honestly I actually prefer at least the description being modified the same way as any other comment. It feels intuitive as I still think of it as a comment.
> 
> Definitely curious if you think there's a better option however, as well as how we can add functionality to edit the issue title in the future 🤔

I got it, make sense, so we can release that in v1.3 and then reorganize this screen in order to fit edit title and body 😄 

Nice job mate!

housseindjirdeh commented on src/components/button.component.js:
```diff
@@ -188,7 +191,12 @@ export class Button extends Component {
         icon={
-          icon && { ...defaultIconStyle, ...sizes[size].iconStyle, ...icon }
+          icon && {
+            ...defaultIconStyle,
```
Extended this since didn't seem like icons were being styled in the component within @machour

housseindjirdeh commented:
> @housseindjirdeh that icon should be black, otherwise it's the intended look.

@machour fixed:

![image](https://user-images.githubusercontent.com/12476932/31098776-d1146a60-a791-11e7-8ddc-1c1c6337576b.png)

housseindjirdeh commented:
@jouderianjr Exactly what I was thinking :)
